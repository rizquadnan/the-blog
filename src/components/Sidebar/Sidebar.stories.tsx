import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { Sidebar } from './Sidebar'

type TSidebar = React.ComponentProps<typeof Sidebar>
export default {
  title: 'Layout/Sidebar',
  component: Sidebar,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta<TSidebar>

const Template: Story<TSidebar> = (args) => <Sidebar {...args} />

export const Default = Template.bind({})
Default.args = {
  selectedNav: 'home',
  onLogout: action('logout'),
}

export const HideUsersNav = Template.bind({})
HideUsersNav.args = {
  hideNav: 'users',
}
