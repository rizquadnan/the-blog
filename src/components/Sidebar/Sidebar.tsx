import React from 'react'
import { Flex, HStack } from '@chakra-ui/layout'
import { VStack } from '@/components/uiKit/VStack'
import { Link } from '@/components/uiKit/Link'
import {
  HomeIcon,
  PowerIcon,
  SettingsIcon,
  UserGroupIcon,
} from '@/components/uiKit/Icons'
import { Button, ButtonProps, useColorModeValue } from '@chakra-ui/react'
import { ColorModeToggle } from '@/components/ColorModeToggle'
import { FEATURE_FLAG } from '@/config/featureFlag'

type TNavItems = 'post' | 'users' | 'account' | 'logout'
type TSidebar = {
  selectedNav: TNavItems
  onLogout: () => void
  hideNav?: TNavItems
}

export function Sidebar(props: TSidebar) {
  const boxShadowColor = useColorModeValue(
    'rgba(149, 157, 165, 0.2)',
    'rgba(149, 157, 165, 0.05)',
  )

  const selectedNavStyles: Partial<ButtonProps> = {
    colorScheme: 'orange',
    variant: 'outline',
    opacity: 1,
  }

  return (
    <Flex
      flexDirection="column"
      justifyContent="space-between"
      minH="100vh"
      padding="32px 24px"
      w="fit-content"
      boxShadow={`${boxShadowColor} 0px 8px 12px`}
    >
      <VStack minW="160px">
        {props.hideNav === 'post' ? null : (
          <Button
            as={Link}
            href="/posts"
            leftIcon={<HomeIcon fontSize="18px" />}
            variant="ghost"
            justifyContent="flex-start"
            {...(props.selectedNav === 'post' ? selectedNavStyles : {})}
          >
            Posts
          </Button>
        )}
        {props.hideNav === 'users' ? null : (
          <Button
            as={Link}
            href="/users"
            leftIcon={<UserGroupIcon fontSize="18px" />}
            variant="ghost"
            justifyContent="flex-start"
            {...(props.selectedNav === 'users' ? selectedNavStyles : {})}
          >
            Users
          </Button>
        )}
      </VStack>
      <VStack>
        {props.hideNav === 'account'
          ? null
          : FEATURE_FLAG.enableMyAccount && (
              <Button
                as={Link}
                href="/my-account"
                leftIcon={<SettingsIcon fontSize="18px" />}
                variant="ghost"
                justifyContent="flex-start"
              >
                My Account
              </Button>
            )}
        {props.hideNav === 'logout'
          ? null
          : FEATURE_FLAG.enableLogout && (
              <Button
                leftIcon={<PowerIcon fontSize="18px" />}
                variant="ghost"
                justifyContent="flex-start"
                onClick={props.onLogout}
              >
                Logout
              </Button>
            )}
        <ColorModeToggle />
      </VStack>
    </Flex>
  )
}

export default Sidebar
