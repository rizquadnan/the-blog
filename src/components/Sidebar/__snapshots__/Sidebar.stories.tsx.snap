// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout/Sidebar Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Posts",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Users",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Dark Mode",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "layout-sidebar--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`Layout/Sidebar HideUsersNav smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Posts",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Dark Mode",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "layout-sidebar--hide-users-nav",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
